### ISSUE TITLE
Improvement on Top Downloaded Apps

### ISSUE THREAD

**Issue by yihangx** [cytoscape/appstore#112]
Now, top downloaded apps show all-time top downloaded apps, which is kind of outdated. It should show top downloaded apps in last 24 months.

Related code:
https://github.com/cytoscape/appstore/blob/dce25f16b3ed735ec1e16bf9c764064ee72f88c4/apps/views.py#L90

https://github.com/cytoscape/appstore/blob/dce25f16b3ed735ec1e16bf9c764064ee72f88c4/apps/views.py#L118

**Comment by qzMalekuz** [comment]
Hi @yihangx ,

I reviewed the download tracking models and noticed that historical download data is stored per release via ReleaseDownloadsByDate.

My understanding is that implementing this would involve:

Filtering ReleaseDownloadsByDate records for the last 24 months

Aggregating count grouped by App (via the Release relationship)

Ordering apps by the aggregated value

Before proceeding, I’d like to confirm:

Should this logic replace the existing downloads field for the “Top Downloaded Apps” section only?

Or should the downloads field itself be recalculated dynamically?

I can prepare a PR using ORM aggregation once the intended behavior is confirmed.

Thanks.

**Comment by coleslaw481** [comment]
> Hi [@yihangx](https://github.com/yihangx) ,
> 
> I reviewed the download tracking models and noticed that historical download data is stored per release via ReleaseDownloadsByDate.
> 
> My understanding is that implementing this would involve:
> 
> Filtering ReleaseDownloadsByDate records for the last 24 months
> 
> Aggregating count grouped by App (via the Release relationship)
> 
> Ordering apps by the aggregated value
> 
> Before proceeding, I’d like to confirm:
> 
> Should this logic replace the existing downloads field for the “Top Downloaded Apps” section only?
> 
> Or should the downloads field itself be recalculated dynamically?
> 
> I can prepare a PR using ORM aggregation once the intended behavior is confirmed.
> 
> Thanks.

Yes it should only be used to show what apps are displayed in the top downloaded apps UI.

thanks,
chris

**Comment by qzMalekuz** [comment]
Hi @coleslaw481 ,

I’ve opened a PR - #119 implementing the 24-month aggregation logic for
Top Downloaded Apps UI using ReleaseDownloadsByDate.

Please let me know if you’d like any adjustments.

Thanks!
